File: standing-app-ng/src/app/model/league.ts
import { User } from "./user";

export class League {
    id: number;
    name: string;
    birthdayBefore: Date;
    year: number;
    managerId: number;
    status: string;
    manager: User;

    constructor(
        id: number = 0,
        name: string = '',
        birthdayBefore: Date = new Date,
        year: number = 0,
        managerId: number = 0,
        status: string = '',
        manager: User = new User()
    ){
        this.id = id;
        this.name = name;
        this.birthdayBefore = birthdayBefore;
        this.year = year;
        this.managerId = managerId;
        this.status = status;
        this.manager = manager;
    }
}

File: StandingsApp/StandingsApp/Models/EF/League.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace StandingsApp.Models.EF
{
    [Table("League")]
    public class League
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(20)]
        public string Name { get; set; }

        [Required]
        public DateTime BirthdayBefore { get; set; }

        [Required]
        public int Year { get; set; }

        [Required]
        public int ManagerId { get; set; }

        [Required]
        public string Status { get; set; }

        public virtual User? Manager { get; set; }

    }
}
